src/coreclr/System.Private.CoreLib/src/Internal/Runtime/InteropServices/ComponentActivator.CoreCLR.cs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace Internal.Runtime.InteropServices
+{
+    public static partial class ComponentActivator
+    {
+        // This hook for when GetFunctionPointer is called when the feature is disabled allows us to
+        // provide error messages for known hosting scenarios such as C++/CLI.
+        private static void OnDisabledGetFunctionPointerCall(IntPtr typeNameNative, IntPtr methodNameNative)
+        {
+            if (!OperatingSystem.IsWindows())
+                return;
+
+            // Check for the exact type and method name used by ijwhost - see src/native/corehoste/ijwhost/ijwhost.cpp
+            if (Marshal.PtrToStringUni(methodNameNative) == "LoadInMemoryAssemblyInContext"
```

Comment:
This is leaking the converted strings. You may want to instead convert the managed string to byte[] that gets garbage collected, compare it with the incoming string as Span; and add a note to switch it over to https://github.com/dotnet/csharplang/blob/main/proposals/utf8-string-literals.md once it becomes available in the compiler.

Reply:
Sorry, I am not following and might need this spelled out for me. The return of `Marshal.PtrToStringUni` is being leaked here? And the suggestion is to turn the string it returns into a byte[]? I'm also missing how I should use the UTF-8 string literal for comparing against the incoming UTF-16 string?

Reply:
Ah sorry. I got confused. There is no memory leak here.